### PR TITLE
Add container image building to CI config

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR

- updates CI configuration to use `architect-orb` to 0.13.0
- adds container image publishing at [`quay.io/giantswarm/kubectl-gs`](https://quay.io/repository/giantswarm/kubectl-gs)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ## [Unreleased]
 
+### Added
+- Publish a container image of kubectl-gs as quay.io/giantswarm/kubectl-gs
```
Comment (kopiczko) [inline CHANGELOG.md:11]:
```suggestion
- Start publishing a container image of kubectl-gs as giantswarm/kubectl-gs
```

It will be synced by crsync.

**Review by kopiczko — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,9 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ## [Unreleased]
 
+### Added
+- Start publishing a container image of kubectl-gs as giantswarm/kubectl-gs
```
Comment (marians) [inline CHANGELOG.md:11]: Should be `as quay.io/giantswarm/kubectl-gs?, right?

Comment (kopiczko) [reply CHANGELOG.md:11]: Longterm we decided dockerhub to be the source of truth. `crsync` will move the image there.

@JosephSalisbury - correct me if I'm wrong.

Comment (JosephSalisbury) [reply CHANGELOG.md:11]: @kopiczko it's the other way around, right? we push everything to `quay` (still), and crysnc moves it to dockerhub and ACR?

Comment (JosephSalisbury) [reply CHANGELOG.md:11]: yeah, see the second diagram here: https://intranet.giantswarm.io/docs/architecture-specs-adrs/specs/registry-mirror/#how-are-images-synchronised-between-registries

Comment (kopiczko) [reply CHANGELOG.md:11]: Yes but we want dockerhub to be the SoT eventually. So why would be advertise quay?

Comment (JosephSalisbury) [reply CHANGELOG.md:11]: ah, with you. yeah, `giantswarm/kubectl-gs` makes sense as our source of truth

**Pull request merged**